dev/release/README.md

```diff
@@ -23,55 +23,85 @@ DataFusion typically has major releases around once per month, including breakin
 
 Patch releases are made on an adhoc basis, but we try and avoid them given the frequent major releases.
 
-## Branching Policy
+## Release Process Overview
 
-- When we prepare a new release, we create a release branch, such as `branch-37` in the Apache repository (not in a fork)
-- We update the crate version and generate the changelog in this branch and create a PR against the main branch
-- Once the PR is approved and merged, we tag the rc in the release branch, and release from the release branch
-- Bug fixes can be merged to the release branch and patch releases can be created from the release branch
+New development happens on the `main` branch.
+Releases are made from branches, e.g. `branch-37` for the `37.x.y` release series.
 
-#### How to backport (add changes) to `branch-*` branch
+To prepare for a new release series, we:
 
-If you would like to propose your change for inclusion in a release branch for a
-patch release:
+- Create a new release branch from `main`, such as `branch-37` in the Apache repository (not in a fork)
+- Continue merging new features changes to `main` branch
+- Prepare the release branch for release:
+  - Update version numbers in `Cargo.toml` files and create `CHANGELOG.md`
+  - Backport any needed changes from `main` to the release branch
+- When the code is ready, we create GitHub tags and create release candidate (rc) artifacts from the release branch.
+- After the release is approved, we publish to crates.io, the ASF distribution servers, and GitHub tags.
+
+For patch releases, we backport changes from `main` to the release branch as
+needed and repeat the release process as needed.
```

Comment:
Another step in my mind is to forward port, like this PR: https://github.com/apache/datafusion/pull/18057/files